apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/post-content/edit.js

```diff
@@ -11,94 +13,137 @@ import { get } from 'lodash';
+		const showPostTitleBeforeContent = this.props.isFullSitePage && ! forceDefaultPostTitle;
+		document
+			.querySelector( '#editor' )
+			.classList.toggle( 'show-post-title-before-content', showPostTitleBeforeContent );
+	}
 
-	const showToggleButton = ! isEditing || !! selectedPost;
-	const showPlaceholder = isEditing || ! selectedPost;
-	const showPreview = ! isEditing && !! selectedPost;
+	render() {
+		const { attributes, isEditing, isFullSitePage, selectedPost } = this.props;
+		const { align } = attributes;
 
-	return (
-		<Fragment>
-			{ showToggleButton && (
-				<BlockControls>
-					<Toolbar>
-						<IconButton
-							className={ classNames( 'components-icon-button components-toolbar__control', {
-								'is-active': isEditing,
-							} ) }
-							label={ __( 'Change Preview' ) }
-							onClick={ toggleEditing }
-							icon="edit"
-						/>
-					</Toolbar>
-				</BlockControls>
-			) }
-			<div
-				className={ classNames( 'post-content-block', {
-					[ `align${ align }` ]: align,
-				} ) }
-			>
-				{ showPlaceholder && (
-					<Placeholder
-						icon="layout"
-						label={ __( 'Content Slot' ) }
-						instructions={ __( 'Placeholder for a post or a page.' ) }
-					>
-						<div className="post-content-block__selector">
-							<div>{ __( 'Select something to preview:' ) }</div>
-							<PostAutocomplete
-								initialValue={ get( selectedPost, [ 'title', 'rendered' ] ) }
-								onSelectPost={ onSelectPost }
-								postType={ [ 'page', 'post' ] }
+		const isTemplatePostType = 'wp_template' === fullSiteEditing.editorPostType;
+		const showToggleButton = isTemplatePostType && ( ! isEditing || !! selectedPost );
+		const showPlaceholder = isTemplatePostType && ( isEditing || ! selectedPost );
+		const showPreview = isTemplatePostType && ! isEditing && !! selectedPost;
+		const showInnerBlocks = ! isTemplatePostType;
+		const showPostTitle = ! isTemplatePostType && isFullSitePage;
+
+		return (
+			<Fragment>
+				{ showToggleButton && (
+					<BlockControls>
+						<Toolbar>
+							<IconButton
+								className={ classNames( 'components-icon-button components-toolbar__control', {
+									'is-active': isEditing,
+								} ) }
+								label={ __( 'Change Preview' ) }
+								onClick={ this.toggleEditing }
+								icon="edit"
 							/>
-							{ !! selectedPost && (
-								<a href={ `?post=${ selectedPost.id }&action=edit` }>
-									{ sprintf( __( 'Edit "%s"' ), get( selectedPost, [ 'title', 'rendered' ], '' ) ) }
-								</a>
-							) }
-						</div>
-					</Placeholder>
-				) }
-				{ showPreview && (
-					<RawHTML className="post-content-block__preview">
-						{ get( selectedPost, [ 'content', 'rendered' ] ) }
-					</RawHTML>
+						</Toolbar>
+					</BlockControls>
 				) }
-			</div>
-		</Fragment>
-	);
-} );
+				<div
+					className={ classNames( 'post-content-block', {
+						[ `align${ align }` ]: align,
+					} ) }
+				>
+					{ showPostTitle && <PostTitle /> }
```

Comment:
Nice idea @mmtr !

Also non-blocking but noting that we'll need to address focus later to allow for permalink modification.

![permalink](https://user-images.githubusercontent.com/1270189/60138581-bcfe9900-975f-11e9-8102-637229775496.gif)